### ISSUE TITLE
Flow: Reduce phlare.* update times

### ISSUE THREAD

**Issue by ptodev** [grafana/agent#2871]
Evaluating the config for `phlare.*` components seems to take a lot longer than for the other components. This causes the "agent_component_evaluation_seconds" histogram to log much higher readings than usual. We should look into whether this update time can be reduced.

**Comment by tpaschalis** [comment]
cc @cyriltovena for more context, as this looks like it has to do with https://github.com/grafana/agent/commit/96f8f80113021cb7f22f142a73a005ce5274c8ae.

At a quick glance, and following the Update method where we apply the scraper's config, and reload every active target, the stop() method closes the graceShut channel, but this might be received only after the scrape has been finished.

https://github.com/grafana/agent/blob/2528c464478fddaa49e04d2617f4bfad36574ecd/component/phlare/scrape/scrape_loop.go#L195-L200

I don't know if there's another issue at play here and I don't have a solution to propose yet, but I'd like to see whether there's an easy way to reload in the background.

**Comment by rfratto** [comment]
We're seeing `phlare.scrape` take up to 17 _minutes_ to evaluate: 

![image](https://user-images.githubusercontent.com/630212/221626646-fb37d51e-28b8-4e16-8816-9126522d316c.png)

This is way too long, and it prevents other components from being able to update for 17 minutes. 

cc @cyriltovena
